store/sqlstore/upgrade.go

```diff
@@ -1230,6 +1230,11 @@ func upgradeDatabaseToVersion538(sqlStore *SqlStore) {
 // last viewed at time of the the thread as the last viewed at time
 // of the channel
 func fixCRTThreadCountsAndUnreads(sqlStore *SqlStore) {
+	var system model.System
+	if err := sqlStore.GetMaster().SelectOne(&system, "SELECT * FROM Systems WHERE Name = 'CRTThreadCountsAndUnreadsMigrationComplete'"); err == nil {
+		return
+	}
```

Comment:
If CRT is enabled, viewing a channel should NOT mark a thread as read. This migration will mark some threads in channels as read, every time it is run. This is a problem on daily. We leverage the systems table to ensure this migration is done only once. 

We might have to toggle this once after https://mattermost.atlassian.net/browse/MM-36737 is done.